### DISCUSSION TITLE
SQL member request

### DISCUSSION THREAD

**Discussion by meteos77** [textbrowser/biblioteq#354] [Q&A] [answered]
I have a question about an SQL query that I can't formulate.

My goal is to get the answer to the following question: To know the number of new members in 2023 under 14 years of age ?

`SELECT COUNT(*) AS NUMBER, SUBSTR(membersince,7,4) AS ANNEE_INSCRIPTION, last_name, SUBSTR (dob,7,4) AS ANNEE_ANNIVERSAIRE FROM member WHERE ANNEE_INSCRIPTION = ('2023') AND ANNEE_ANNIVERSAIRE >= (SELECT SUBSTR(DATE('now', '-14 year'), 1,4))`

I'd like to keep only the NUMBER column, for the moment I'm filtering with python but I think SQL is powerful enough to do it?

Can you put me on the track,  Thank you in advance

**Comment by textbrowser** [comment]
```select julianday(date('now')) - julianday(substr(dob, 7) || '-' || substr(dob, 1, 2) || '-' || substr(dob, 4, 2)) from member;```

**Comment by textbrowser** [comment]
Divide that by 365 and that will be an approximate number of years.

**Comment by textbrowser** [comment]
```select (julianday(date('now')) - julianday(substr(dob, 7) || '-' || substr(dob, 1, 2) || '-' || substr(dob, 4, 2))) / 365.0 from member;```

**Comment by meteos77** [comment, marked as answer]
After a lot of trial and error, I found my request.
Thanks for your lead ... I'm looking for unnecessary complication
`SELECT count(*) AS NUMBER FROM member WHERE SUBSTR(membersince,7,4) = ('2024') AND SUBSTR(dob,7,4) >= (SELECT SUBSTR(DATE('now', '-14 year'), 1,4))`

**Comment by textbrowser** [comment]
Nice yob.
